src/Algebra/Properties/Monoid/Reasoning.agda

```diff
@@ -0,0 +1,68 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Equational reasoning for monoids
+-- (Utilities for identity and cancellation reasoning, extending semigroup reasoning)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Monoid)
+
+module Algebra.Properties.Monoid.Reasoning {o ℓ} (M : Monoid o ℓ) where
+
+open Monoid M
+    using (Carrier; _∙_; _≈_; setoid; isMagma; semigroup; ε; sym; identityˡ
+    ; identityʳ ; ∙-cong; refl; assoc; ∙-congˡ; ∙-congʳ; trans)
+open import Relation.Binary.Reasoning.Setoid setoid
+open import Algebra.Properties.Semigroup.Reasoning semigroup public
+
+module Identity {a : Carrier } where
+    id-unique : (∀ b → b ∙ a ≈ b) → a ≈ ε
+    id-unique b∙a≈b = trans (sym (identityˡ a)) (b∙a≈b ε)
+
+    id-comm : a ∙ ε ≈ ε ∙ a
+    id-comm = trans (identityʳ a) (sym (identityˡ a))
+
+    id-comm-sym : ε ∙ a ≈ a ∙ ε
+    id-comm-sym = sym id-comm
+
+open Identity public
+
+module IntroElim {a b : Carrier} (a≈ε : a ≈ ε) where
```

Comment:
My instincts tell me:
* `a` can be implicit, and covered by a `variable` declaration
* `b` should be explicit